### PR TITLE
stats/opentelemetry: Cleanup OpenTelemetry API's before stabilization

### PR DESCRIPTION
**Pull request by zasweq**
This PR cleans up some OpenTelemetry API's before stabilization.

RELEASE NOTES:
* stats/opentelemetry: Introduce new APIs to enable OpenTelemetry instrumentation for metrics on servers and clients

### CODE REVIEW

File: stats/opentelemetry/opentelemetry.go [hunk 1]
```diff
@@ -57,27 +62,28 @@ type Options struct {
...
+
+	// MethodAttributeFilter is a function that determines whether to record the
+	// method name of RPCs as an attribute, or to bucket into "other". Take care
+	// to limit the values allowed, as allowing too many will increase
+	// cardinality and could cause severe memory or performance problems. On
+	// Client Side, to record the method name along with metric instead of
+	// bucketing into "other pass a grpc.StaticMethodCallOption as a call option
+	// into Invoke or NewStream.
+	//
 	// This only applies for server side metrics.
```
Comment (dfawley) [inline stats/opentelemetry/opentelemetry.go:82]:
```suggestion
	// cardinality and could cause severe memory or performance problems.
	//
	// This only applies for server-side metrics.  For clients, to record
	// the method name in the attributes, pass grpc.StaticMethodCallOption
	// to Invoke or NewStream. Note that when using protobuf generated
	// clients, this CallOption is included automatically.
```

File: stats/opentelemetry/opentelemetry.go [hunk 2]
```diff
@@ -57,27 +62,28 @@ type Options struct {
...
 	MethodAttributeFilter func(string) bool
 
-	// OptionalLabels are labels received from LB Policies that this component
-	// should add to metrics that record after receiving incoming metadata.
+	// OptionalLabels specifies a list of optional labels that will be included
+	// on any metrics that have corresponding optional labels configured.
```
Comment (dfawley) [inline stats/opentelemetry/opentelemetry.go:86]:
```suggestion
	// OptionalLabels specifies a list of optional labels to enable on any
	// metrics that support them.
```

**Review by dfawley — COMMENTED** [review]
LGTM, just some minor wording suggestions.  Please check out the edits-as-commits and suggested changes.

File: stats/opentelemetry/opentelemetry.go (hunk 1 above)
Comment (zasweq) [reply stats/opentelemetry/opentelemetry.go:82]: Thanks, done.

File: stats/opentelemetry/opentelemetry.go (hunk 2 above)
Comment (zasweq) [reply stats/opentelemetry/opentelemetry.go:86]: Yeah I couldn't figure out a way to tersely convey it's just the subset that are "supported" that are enabled. (i.e. it's not always the full set added optionally, but the ones that are actually also configured. Changed to your suggestion.

**Review by dfawley — APPROVED** [review]

**Pull request merged**